k.makoyid/oop_code/Model_train.py

```diff
@@ -0,0 +1,88 @@
+import numpy as np
+
+
+class ModelTrainer:
+    def __init__(self, model, X_train, Y_train, X_test, Y_test, initial_lr, batch_size, lambda_wd):

```

Comment:
1) add default values for  initial_lr, batch_size, lambda_wd (the optimal ones which you obtained).
2) add lr_schedule either as array [epoch_num_1, epoch_num_2, epoch_num_3 ,..] or as array of pairs [(epoc_num_1, factor_1), (epoch_num_2, factor_2), (epoch_3, factor_3), ...], where , factor__i is learning rate factor. Set some default value for lr_schedule